### ISSUE TITLE
Add methods for object deletion, replacement, etc.

### ISSUE THREAD

**Issue by goodmami** [xigt/xigt#39]
There are currently only methods for adding new objects to collections (`append()`, `extend()`, `insert()`) or for deleting all (`clear()`). What is lacking is methods for deleting individual objects, or for setting them to some new value. Python has a `remove()` method for lists that removes the first matching object, and a `pop()` method that removes the last object, or the object at some index if given. Javascript has `pop()` and `shift()`, which remove objects from the end and start of an Array, respectively. Removing an object in the middle of the list is more complicated: `ary.splice(startPos, 1)`. Removing an object by value is even more complicated: `ary.splice(ary.indexOf(obj), 1)`). However the `splice()` method is nice in that you can remove any number of contiguous objects. For removing objects, the more Pythonic `remove()` seems like a good fit, although its semantics could be more like the `get()` method:

- `remove(x: int)` = remove object at index `x`
- `remove(x: str)` = remove object with id `x`

Neither of these is like `list.remove()` because it doesn't take the object to remove itself as the argument, but doing so might not be a bad idea, either.

Methods for setting (as opposed to adding) objects could be useful. Lists can set objects by index (`some_list[3] = new_val`), and dicts can set values by key (`some_dict[key] = new_val`), so maybe just defining a `__setitem__()` function would be good.

**Comment by goodmami** [comment]
RE `remove()`, if we try to stay more Pythonic, it should actually just remove the object that is its argument. For indices or keys, use `del container[index_or_id]`

**Comment by goodmami** [comment]
Ok, so I've implemented and tested a solution:
- `container.remove(obj)` where `obj` is the actual object instance to be removed
- `del container[idx_or_id]` where `idx_or_id` can be the object-to-delete's list index or ID key
- `container[idx] = obj` where `idx` is *only* a list index, not an ID key (see below)

For `__setitem__()`, I only allow list indices because it's unclear what to do with ID keys, since the container is really a list and the dict keys are just for convenience. For instance, if a key is given that doesn't match the ID of the object to set, or if a new ID is used:

```python
container['x5'] = Item(id='x6')  # does this get inserted into the dict at x5? x6?
container['x999'] = Item(id='x999')  # assuming this is a new key, do we append to the end of the list?
container['x5'] = Item()  # do we set the Item's ID to x5?
```

Because of these problems, I decided it was best to only allow list indices here (note that in a future version of Xigt, the conflation of keys/indices might be removed entirely, thus changing `__getitem__()` and `__delitem__()` as well; see #41). If one wants to set an object by ID, it can be done more manually:

```python
idx = container.index(container['x5'])
container[idx] = Item(id='x6')
```

This is much more explicit as to what is going on, and more clear as to what the result will be.
